lib/Backend/GlobOpt.cpp

```diff
@@ -9591,13 +9594,29 @@ GlobOpt::TypeSpecializeInlineBuiltInBinary(IR::Instr **pInstr, Value *src1Val, V
             {
                 bool lossy = false;
 
+                this->ToInt32(instr, instr->GetSrc2(), this->currentBlock, src2Val, nullptr, lossy);
+
                 IR::Opnd* src1 = instr->GetSrc1();
-                this->ToFloat64(instr, src1, this->currentBlock, src1Val, nullptr, IR::BailOutPrimitiveButString);
+                int32 valueMin, valueMax;
+                if (src1Val->GetValueInfo()->IsLikelyInt() &&
+                    this->DoPowIntIntTypeSpec() &&
+                    src2Val->GetValueInfo()->GetIntValMinMax(&valueMin, &valueMax, this->DoAggressiveIntTypeSpec()) &&
+                    valueMax >= 0)
```

Comment:
Good catch.

---
In reply to: [57935934](https://github.com/Microsoft/ChakraCore/pull/690#discussion_r57935934) [](ancestors = 57935934)